clients/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemProducer.java

```diff
@@ -481,12 +522,8 @@ public synchronized void start() {
 
     if ((pushType.equals(Version.PushType.STREAM) || pushType.equals(Version.PushType.STREAM_REPROCESSING))
         && hybridStoreDiskQuotaEnabled) {
-      hybridStoreQuotaMonitor = Optional.of(
-          new RouterBasedHybridStoreQuotaMonitor(
-              new D2TransportClient(discoveryResponse.getD2Service(), childColoD2Client),
-              storeName,
-              pushType,
-              topicName));
+      hybridStoreQuotaMonitor =
+          Optional.of(new RouterBasedHybridStoreQuotaMonitor(transportClient, storeName, pushType, topicName));
```

Comment:
That seems like a good refactoring... instantiating multiple clients for these components did not seem ideal. Thanks.